### ISSUE TITLE
No gradient defined for op

### ISSUE THREAD

**Issue by minjac** [tensorflow/tensorflow#37467]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): 
auto entropy_0 = Xlogy(scope_loss.WithOpName("xlogy0"), input_labels_var, out_segmentation);
- OS Platform: Windows7
- VS2015
- Python version: 3.6
- Bazel version: 0.26.0
- TensorFlow version: tensorflow 2.0
- CUDA/cuDNN version: Cuda 10.0 && Cudnn 7.4.1
- GPU : Geforce GTX 1070 8GB:

Hey, I got some error when I using op like Concat and Xlogy, which remind me that I have to add C++ gradient. But as https://github.com/tensorflow/tensorflow/issues/19944#issuecomment-556315768 says, the gradient op is existing, so how should I do to use the gradient op?
Thanks !

**Comment by Saduf2019** [comment]
@minjac 
Please share simple indented stand alone code, for us to replicate the issue faced by you in our local environment.

**Comment by minjac** [comment]
> 
> 
> @minjac
> Please share simple indented stand alone code, for us to replicate the issue faced by you in our local environment.

	Scope scope = Scope::NewRootScope();
	auto input = Placeholder(scope.WithOpName("input"), DT_FLOAT);

	map<string, Output> vars;
	map<string, Output> assigns;
	vars["0"] = Variable(scope.WithOpName("w1"), { 1 }, DT_FLOAT);
	vars["1"] = Variable(scope.WithOpName("w2"), { 1 }, DT_FLOAT);
	assigns["0"] = Assign(scope.WithOpName("assign1"), vars["0"], Input::Initializer(0.f, { 1 }));
	assigns["1"] = Assign(scope.WithOpName("assign2"), vars["1"], Input::Initializer(0.f, { 1 }));

	vector<Output> weights;
	weights.push_back(vars["0"]);
	weights.push_back(vars["1"]);

	auto mul1 = Mul(scope.WithOpName("mul1"), input, vars["0"]);
	auto mul2 = Mul(scope.WithOpName("mul2"), input, vars["1"]);

	auto concat = Xlogy(scope.WithOpName("xlogy"), mul1, mul2);

	auto label = Placeholder(scope.WithOpName("label"), DT_FLOAT);
	auto loss = SquaredDifference(scope.WithOpName("label"), concat, label);

	vector<Output> grad_outputs;
	TF_CHECK_OK(AddSymbolicGradients(scope, { loss }, weights, &grad_outputs));

	int index = 0;
	vector<Operation> out_grads;
	for (pair<string, Output> i : vars)
	{
		string s_index = to_string(index);
		auto adam = ApplyGradientDescent(scope, i.second, 0.01f, { grad_outputs[index] });
		out_grads.push_back(adam.operation);
		index++;
	}
	Tensor train_input(DT_FLOAT, {1});
	train_input.scalar<float>()(0) = 0.1;
	Tensor train_label(DT_FLOAT, {2});
	train_label.scalar<float>()(0) = 0.1;
	train_label.scalar<float>()(1) = 0.2;

	unique_ptr<ClientSession> t_session;
	vector<Output> ops_to_run;
	for (pair<string, Output>i : assigns)
	{
		ops_to_run.push_back(i.second);
	}
	t_session = unique_ptr<ClientSession>(new ClientSession(scope));
	TF_CHECK_OK(t_session->Run(ops_to_run, nullptr));

	vector<Tensor> out_tensors;
	TF_CHECK_OK(t_session->Run({ { input, train_input},{ label, train_label } },{loss},out_grads,&out_tensors));

--------------------------------------------
thanks

**Comment by minjac** [comment]
> 
> 
> @minjac
> I ran your code on nightly and [face this error](https://colab.sandbox.google.com/gist/Saduf2019/aa6a1895a9c279c80022fd124bfd6307/37467.ipynb)

I had not faced this error, because I ran my code on VS2015 by using tensorflow_cc.dll.

**Comment by saxenasaurabh** [comment]
Gradient for `Xlogy` is not implemented yet. But you are welcome to send a PR! You can take a look at the existing C++ gradient implementations [here](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/cc/gradients) and the python gradient for `Xlogy` [here](https://github.com/tensorflow/tensorflow/blob/311792b00583b90e507e2c7f8365b4c5ff728874/tensorflow/python/ops/math_grad.py#L660).

**Comment by sushreebarsa** [comment]
@minjac It looks like you are using an older Version of Tensorflow . Many bugs have been fixed in the latest version. Could you please execute your code using Latest Version 2.5 and let us know if the issue still persists? Thanks!
